tests/test_project_structure.py

```diff
@@ -35,3 +70,50 @@ def assert_file_not_contains(self, filename, pattern):
         with open(filename, 'rb', 0) as file, mmap.mmap(file.fileno(), 0, access=mmap.ACCESS_READ) as content:
             if content.find(bytes(pattern, 'utf-8')) != -1:
                 self.fail(f"File {filename} contians illegal pattern - {pattern}")
+
+    def test_providers_modules_should_have_tests(self):
+        # TODO: Should we extend this test to cover other directories?
+        expected_test_files = glob.glob(f"{ROOT_FOLDER}/airflow/providers/**/*.py", recursive=True)
+        # Make path relative
+        expected_test_files = (os.path.relpath(f, ROOT_FOLDER) for f in expected_test_files)
+        # Exclude example_dags
+        expected_test_files = (f for f in expected_test_files if "/example_dags/" not in f)
+        # Exclude __init__.py
+        expected_test_files = (f for f in expected_test_files if not f.endswith("__init__.py"))
+        # Change airflow/ to tests/
+        expected_test_files = (
+            f'tests/{f.partition("/")[2]}'
+            for f in expected_test_files if not f.endswith("__init__.py")
+        )
+        # Add test_ prefix to filename
+        expected_test_files = (
+            f'{f.rpartition("/")[0]}/test_{f.rpartition("/")[2]}'
+            for f in expected_test_files if not f.endswith("__init__.py")
+        )
+
+        current_test_files = glob.glob(f"{ROOT_FOLDER}/tests/providers/**/*.py", recursive=True)
+        # Make path relative
+        current_test_files = (os.path.relpath(f, ROOT_FOLDER) for f in current_test_files)
+        # Exclude __init__.py
+        current_test_files = (f for f in current_test_files if not f.endswith("__init__.py"))
+
+        expected_test_files = set(expected_test_files)
+        current_test_files = set(current_test_files)
+
+        missing_tests_files = expected_test_files - expected_test_files.intersection(current_test_files)
+        self.assertEqual(missing_tests_files, MISSING_TEST_FILES)
```

Comment:
I think `missing_tests_files <= MISSING_TEST_FILES` is the proper condition, as `missing_tests_files` can be subset of `MISSING_TEST_FILES`. In case someone added a missing test this test should pass and `test_keep_missing_test_files_update` should fail. Otherwise, we are testing the same thing two times.

Reply:
I think it's good with equals. The goal is - similar as with pylint_todo.txt to go with this list to empty. So when someone adds test file, the test will fail and it will be the right time for the person who adds this file to remove it from the list.

Reply:
I agree but we have two tests. One that checks missing files (this one) and second one that checks if someone adjusted the `MISSING_TEST_FILES`.

Reply:
Thanks. Updated.

Reply:
Indeed. that's nicer now!